app/ide-desktop/lib/dashboard/src/authentication/src/dashboard/components/dashboard.tsx

```diff
@@ -139,95 +222,34 @@ const dashboardContainer = (props: DashboardProps) => {
-        const newProjectList = [...currProjectList];
-        newProjectList[projectItemIndex]!.state.type =
-          backend.ProjectState.closed;
-        return newProjectList;
-      });
-    };
-
-    itemsTable = projectsList.map((item, index) => {
-      return (
-        <tr
-          key={item.projectId}
-          className="transition duration-300 ease-in-out hover:bg-gray-100"
-        >
-          <td className="px-6 text-left flex items-center align-middle whitespace-nowrap border border-solid border-l-0 border-r-1 border-t-0 border-b-0">
-            <ProjectActionButton
-              project={item}
-              onOpen={() => {
-                setProjectOpen(index);
-              }}
-              onOpenStart={() => {
-                setProjectOpening(index);
-              }}
-              onClose={() => {
-                setProjectClosed(index);
-              }}
-            />
-            {item.name}
-          </td>
-          <td className="px-6 border border-solid border-l-0 border-r-1 border-t-0 border-b-0">
-            2022-08-10, 13:30
-          </td>
-          <td className="px-6 border border-solid border-l-0 border-r-1 border-t-0 border-b-0">
-            aa
-          </td>
-          <td className="px-6 border border-solid border-l-0 border-r-1 border-t-0 border-b-0">
-            aa
-          </td>
-          <td className="px-6 border border-solid border-l-0 border-r-1 border-t-0 border-b-0">
-            aa
-          </td>
-          <td className="px-6 border border-solid border-l-0 border-r-1 border-t-0 border-b-0">
-            aa
-          </td>
-          <td className="px-6 border border-solid border-l-0 border-r-1 border-t-0 border-b-0"></td>
-          <td className="px-6 border border-solid border-l-0 border-r-1 border-t-0 border-b-0"></td>
-        </tr>
-      );
+  const setProjectState = (
+    projectItemIndex: number,
+    state: backend.ProjectState
+  ): void => {
+    setProjectsList((currProjectList) => {
+      const newProjectList = [...currProjectList];
+      newProjectList[projectItemIndex]!.state.type = state;
+      return newProjectList;
     });
-  }
+  };
 
   return (
     <>
       <Templates onChange={handleCreateProject} />
-      <table className="items-center w-full bg-transparent border-collapse">
-        <thead>
-          <tr>{tableHeaders}</tr>
-        </thead>
-        <tbody>{itemsTable}</tbody>
-      </table>
+      <Table<backend.Project>
+        items={projectsList}
+        getKey={(proj) => proj.projectId}
+        placeholder={
+          <>
+            You have no project yet. Go ahead and create one using the form
+            above.
+          </>
+        }
+        columns={ALL_COLUMNS.map(column => ({ id: column, name: COLUMN_NAME[column], render: renderers[column] }))}
+      />
       <button onClick={signOut}>Log out</button>
     </>
   );
 };
 
-export default withRouter(dashboardContainer);
+export default withRouter(Dashboard);
```

Comment:
hmm, I was sure our prettier was configured to remove semicolons in our codebase. Do we need to use them?

Reply:
this folder was excluded because jsx and tsx weren't included in the prettier config - i think it's a bit to messy to change that right here, so it might be a better idea to wait till #5803 and #5804 are merged before doing a commit/PR to prettier all the files under `dashboard/`?